### ISSUE TITLE
Summarise your readings

### ISSUE THREAD

**Issue by swangcs** [swangcs/BusGPS#6]
Try to summarise your [readings](https://github.com/swangcs/BusGPS/wiki/Resources) in the following aspects:
- research problem
  - e.g. predict bus arrival time? GPS data processing?
- model used
  - e.g. SVM? LR? RNN? Kernel?
- dataset used
  - e.g. Dublin? Rio? One bus line? One month?
- results
  - e.g. accuracy? computational performance?
- **limitations**
  - very important
  - i.e. the weakness of this work that our work intends to fill the gap

**Comment by MrTornado24** [comment]
Predicting arrival times of buses using real-time GPS measurements, 2012
a. Problem: predict bus arrival time
b. dataset used: 100 trajectories of the Dublin 46A  June 2011.
c. Model used: Kernel, Delay, KNN, LR
d. Results: 
Accuracy: Kernel best
Computational performance: Linear to training size, reduced by intercepting
Limitation and weakness: 
better balance between accuracy and compuatational performance
How to get CI
How to integrate additional regression variables

**Comment by swangcs** [comment]
@MrTornado24 please rephrase your writings in markdown style for a proper representation

**Comment by swangcs** [comment]
@MrTornado24 also check details about the KNN implementation by reading the "Coffey et al. 2011" paper

**Comment by MrTornado24** [comment]
@swangcs 
[Time of arrival predictability horzions for public bus routes](https://www.researchgate.net/profile/Francesco_Calabrese2/publication/239761081_Time_of_arrival_predictability_horizons_for_public_bus_routes/links/54cf541f0cf29ca810fde1d9.pdf)
- The implementation of KNN
  - Find similar trips. Given that a bus B has travelled n meters along its route in m seconds. 
    Compare m to the time taken for all historic bus trips to reach n meters and find k trips 
    with minimum difference. Reference to 3.1 of paper.
- Prediction error
  - We find k should be variable over distance to keep the minimum error. Reference to 3.2and 3.3 of paper.
- Optimal k and the predictability horizon
  - For long term prediction, k should approximates to ∞, and for short, k should be distinct. 
  - Optimal k can achieve upper bound of prediction quality, in term of prediction horizon.
Reference to 3.4 and 3.5.
- Question
  - Figure 6. If the y-axis is confidence interval, why it says "The best one can do is take k = ∞ "? We can see the confidence interval reaches the top when k is 1, not ∞(may be ∞, but it is a trend)
  - Figure 7. Why we choose "Time since intinial prediction" as x-axis? How to plot this figure?
  - Is k different for different days of a week or weather? If true, is it could be an alternative method to Kernel Regression because it can reduce the training set and resulting complexity?

**Comment by swangcs** [comment]
@MrTornado24 
Good summary and questions!
- I think the whole 3.5 section is confusing, including Figure 6, which is even lack of explanation for x and y-axis.
- For "Figure 7", I think the author just want to show the prediction error decreases with the gain of more certainties, that is, for a single trip, the more trajectory traveled, the better predict results for the last stop.
- I think k=10 for all cases. Tuning such parameters is a good way to achieve the trade-off between computation complexity and accuracy. But we probably also need to automate this tuning process.

by the way, I think this paper is also worth reading for you @Ruixinhua especially for 2.1 section "data cleansing process".

**Comment by MrTornado24** [comment]
@swangcs 
[Bus Travel Time Predictions Using Additive models](https://arxiv.org/pdf/1411.7973.pdf)
- Preliminaries
  - Motivating Data: 
    1. Each GPS data point: (longtitude, latitude), date, time stamp, bus ID and route ID.
    2. Time peroid: 09/26/2013 to 01/09/2014
    3. Combined with GTFS
  - Data Normalization 
    1. Map 2-D coordinates into 1-D distance.
    2. Normalize the time stamps by interpolating and plot **cumulative space-time trajectories**.
- Proposed Solution
  - BAM
    1. Expression: Tij =β0 +f1(distij)+f2(timei)+f3(distij,timei)+εij, i = 1,...,n, and j = 1,...,mi, where β0, and εij represent an overall model mean and error term, respectively. Pay attention to the specific functions of f1, f2 and f3. 
    2. Performance: adjusted R^2 of the model was 0.903
  - EAM
    1. Expression: Tij = β0 + β1 · weekendi + f1(distij , weekendi) +β ·Tlast +f (time)+f (dist,time)+ε. Add the indicator variable *weekend* (bus occured on a weekday or the weekend)
    2. Performance: adjusted R^2 is 0.919
  - AMM
    1. Expression: See on the paper. Add the random intercept term b0i. 
    2. Performance: adjusted R^2 is 0.968
- Experiments
  - Experimental data: specific four routes
  - Setup: random selection of 14 days as test data -> different size of historical data -> predictions for each bus in test set from every bus stop until end of route -> calculate and analyze prediction errors
  - Evaluation Measures:  *mean absolute relative error* and distribution of absolute errors over prediction distance
  - Implemented Methods and results
    1. SVM: The worst of all.
    2. >the AMM statistically outperformed all methods for all routes and in all distance bins (except for the 14km bin on route 627 where no difference existed between EAM and AMM). In the first distance bin, [0, 1), the Kernel Regression method outperformed both BAM and EAM at all routes except for route 603 (where no statistical difference existed). However, in all other distance bins the two Additive Models statistically outperformed the Kernel Regression. 
    3. >Kernel Regression outperformed BAM, and EAM in the first distance bin suggests that the Additive Models tend to put more priority on minimizing error in later parts, when it is in fact larger, at the expense of short term predictions. Perhaps this may be fixed by placing more knots at the beginning of the route or through additional features.
 - Conclusion
   - Model cumulative travel time as a sum of linear terms and smooth functions of predictor variables. 
   - Advantage: interpretability and flexibility in modeling non-linear relationships. Do not need discretization or interpolation except when normalising the time stamp.
 - Questions
   - I am learning cubic regression spline basis functions, tensor product smooths and other important concepts mentioned in this paper.

**Comment by MrTornado24** [comment]
[Learning to Predict Bus Arrival Time From Heterogeneous Measurements via Recurrent Neural Network](https://sci-hub.tw/10.1109/TITS.2018.2873747)
- Problem
  - Using RNN with LSTM to achieve non_linear bus arrival time prediction.
  - Comparison with regression-based, the filter based and the searching-based method.
  - one-hot coding is introduced to fuse heterogeneous information into a unified vector space.
- Data set and preparation
  - Including dynamic data(GPS) and static data(Location of stops and intersections)
  - Mapping data(dynamic and static) into a spatio-temporal trace.
- RNN with LSTM for Prediction
  - Padding spatial, temporal and static measurements into a vector to show information of stop pk: Xk =[t0,tk,rk,dk,dk+1]T. dk: location of the passed stop; dk+n: position of the ahead stop; tk: arrival time; t0: departure time of a bus at the start stop; rk: the number of routes marked by stop.
  - Building LSTM structure(including input gate, output gate and forget gate for memory block)
  -  Loss function for training.
  -  Recursive prediction including feature coding and usage of prediction model. The vector should be expanded by one-hot code specified to different measurements.
- Experiment and discussion
  - Indicators: MAE, RMSE and MAPE. 
  - Comparison: 
    > In Table. III, LSTM RNN achieves the smallest MAE, RMSE, MAPE for all groups but the “<10 KM” one for MAPE. The performances of LSTM RNN slightly vary with respect to the length of a route. For instance, LSTM RNN achieves 1.23 and 1.19 MAEs in the “10−15 KM” and the “>15 KM” groups, respectively.
    > KFP [15] achieves the worst performance than the other methods. 
    > As expected, the search- based approach tends to achieve inferior results when traffic conditions cannot re-occur for the current prediction. Interest- ingly, the regression-based approach.
    > Principally, the more number of time steps LSTM RNN observes, the more accurate both the states ck and hk are. 
- Conclusion for RNN with LSTM
  - More physically accurate Interpolated points: need more meaningful physically road segments rather than just increasing the sampling rate.
  - More heterogeneous measurements. 
    > such as the automatic passenger counter, automated fare collection, weather, and traffic conditions. Moreover, considering bus routes essentially affect each other, how to incorporate the spatial relationship between bus stops into our prediction model is an interesting direction.
 - Question
   - How is the computation performance of RNN? Why we do not use Markov model? [Reference](https://blog.csdn.net/leochen1983/article/details/64133814)
